### PR TITLE
[MM-55409] Enable revive linter on enterprise code

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
See https://github.com/mattermost/enterprise/pull/1589

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55409

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/oauthproviders/gitlab/gitlab.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func (glu *GitLabUser) getAuthData() string {
 	return strconv.FormatInt(glu.Id, 10)
 }
 
-func (m *GitLabProvider) GetUserFromJSON(c request.CTX, data io.Reader, tokenUser *model.User) (*model.User, error) {
+func (g *GitLabProvider) GetUserFromJSON(c request.CTX, data io.Reader, tokenUser *model.User) (*model.User, error) {
```
Comment (hanzei) [inline server/channels/app/oauthproviders/gitlab/gitlab.go:87]: I was updating the OAuth providers, so I also changed the name here to make more sense.

File: server/.golangci.yml [hunk 2]
```diff
@@ -45,10 +45,6 @@ issues:
       text: "var-naming|error-naming|exported|increment-decrement|error-strings|if-return|unused-parameter|blank-imports|context-as-argument|empty-block"
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/.golangci.yml]: They need a case-by-case review on whether we want to enable them. E.g. `unused-parameter` doesn't seem so useful to me. Some of them need a large amount of changes like `var-naming`or `error-naming`.

My suggestion would be to take the discussion to a thread, decide which linters we want and then open HW tickets for them. What do you think?

[the remark this answers is not in the document]
Comment (agnivade) [reply server/.golangci.yml]: Yeah I know they need a large amount of changes :P That's why I was a bit surprised you wanted to take this up haha

Agree that all of them might not be useful. I am happy to defer this to you Ben. Feel free to do what you think best. :+1:

**Review by agnivade — APPROVED** [review]

File: server/.golangci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply server/.golangci.yml]: https://community.mattermost.com/core/pl/fu6441u54bfhdjaapu8je6kbqy

**Review by isacikgoz — APPROVED** [review]

**Pull request merged**